### PR TITLE
Hpot lon

### PR DESCRIPTION
**Pull request by tccontre**
## New Detections

1. allow_inbound_traffic_in_firewall_rule.yml 
2. enable_rdp_in_other_port_number.yml
3. mailsniper_invoke_functions.yml 
4. allow_inbound_traffic_by_firewall_rule_registry
5. secretdumps_offline_ntds_dumping_tool

### CODE REVIEW

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Allow Inbound Traffic By Firewall Rule Registry :white_check_mark:
- diskUsage: 405504
- runDuration: 7.465s
- scanCount: 12968

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Enable RDP In Other Port Number :white_check_mark:
- diskUsage: 372736
- runDuration: 7.299s
- scanCount: 12968

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

SecretDumps Offline NTDS Dumping Tool :white_check_mark:
- diskUsage: 528384
- runDuration: 1.818s
- scanCount: 18

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Allow Inbound Traffic In Firewall Rule :white_check_mark:
- diskUsage: 167936
- runDuration: 0.992s
- scanCount: 93

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Mailsniper Invoke functions :white_check_mark:
- diskUsage: 188416
- runDuration: 1.184s
- scanCount: 85

**Comment by patel-bhavin** [comment]
Looks good to me!

**Review by patel-bhavin — APPROVED** [review]

File: detections/endpoint/mailsniper_invoke_functions.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+version: 1
+date: '2021-05-19'
+author: Teoderick Contreras, Splunk
+type: batch
+datamodel:
+- Endpoint
+description: This search is to detect known mailsniper.ps1 functions executed in a
+  machine. This technique was seen in some attacker to harvest some sensitive e-mail
+  in a compromised exchange server.
+search: '`powershell` EventCode=4104 Message IN ("*Invoke-GlobalO365MailSearch*",
```
Comment (miskoSplunk) [inline detections/endpoint/mailsniper_invoke_functions.yml:12]: @t-contreras could you please try extending this to a couple of other events, same thing would appear there, but with them we'd ensure coverage:

in EventID 4103 - these same modules would appear in Message field
(see this work in SSA for an "event" sample: https://confluence.splunk.com/display/~tramamurthy/4103+and+4104+mappings+to+Endpoint_process+fields+and+new+fields+addedP

In even 4688: 
These modules would be listed in the Process Command Line field.

File: detections/endpoint/secretdumps_offline_ntds_dumping_tool.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+version: 1
+date: '2021-05-26'
+author: Teoderick Contreras, Splunk
+type: batch
+datamodel:
+- Endpoint
+description: This analytic detects a potential usage of secretsdump.py tool for dumping
+  credentials (ntlm hash) from a copy of ntds.dit and SAM.Security,SYSTEM registrry
+  hive. This technique was seen in some attacker that dump ntlm hashes offline after
+  having a copy of ntds.dit and SAM/SYSTEM/SECURITY registry hive.
```
Comment (miskoSplunk) [inline detections/endpoint/secretdumps_offline_ntds_dumping_tool.yml:12]: @t-contreras please see if you can add/reuse/extend this with the material we had before about secrets dumping in:

credential_dumping_via_copy_command_from_shadow_copy.yml
credential_dumping_via_symlink_to_shadow_copy.yml

ssa___assess_credential_strength_via_dsinternals_modules.yml
ssa___attempted_credential_dump_from_registry_via_reg_exe.yml
ssa___credential_extraction_dsinternals_conversion_modules.yml
ssa___credential_extraction_dsinternals_modules.yml
ssa___credential_extraction_fgdump_cachedump_s_option.yml
ssa___credential_extraction_fgdump_cachedump_v_option.yml
ssa___credential_extraction_getaddbaccount_from_dump.yml
ssa___credential_extraction_lazagne_command_options.yml
ssa___credential_extraction_mimikatz_modules.yml
ssa___credential_extraction_ms_debuggers_kernel_peek.yml
ssa___credential_extraction_ms_debuggers_z_option.yml
ssa___credential_extraction_powersploit_modules.ym
ssa___detect_dump_lsass_memory_using_comsvcs.yml

**Comment by miskoSplunk** [comment]
@t-contreras  Everything else but these 2 comments looks good to me to merge...

File: detections/endpoint/secretdumps_offline_ntds_dumping_tool.yml (hunk 2 above)
Comment (tccontre) [reply detections/endpoint/secretdumps_offline_ntds_dumping_tool.yml:12]: Hi @miskoSplunk  thanks for this suggestion. I will check further on this and see if I can improve it. :)

File: detections/endpoint/mailsniper_invoke_functions.yml (hunk 1 above)
Comment (tccontre) [reply detections/endpoint/mailsniper_invoke_functions.yml:12]: @miskoSplunk thanks for this feedback. will take a look on those events and Ill implement that also if possible ;) thanks again

**Pull request merged**